src/profile/ProfileForm.tsx

```diff
@@ -113,242 +130,270 @@ export const ProfileFormConsumer: React.FC<ProfileConsumerFormProps> = (
-              maxLength: {
-                value: 254,
-                message: "USER.EMAIL.TOO_LONG.254",
-              },
-              pattern: {
-                value: /\S+@\S+\.\S+/,
-                message: "USER.EMAIL.INVALID",
-              },
-            })}
-          />
-        </Col>
-      </Form.Group>
-      <Form.Group as={Row} controlId="writerName">
-        <Form.Label column sm={2}>
-          Writer Name:
-        </Form.Label>
-        <Col sm={10}>
-          <Field
-            errors={props.context.errors}
-            type="text"
-            name="writer_name"
-            ref={props.context.register}
-          />
-        </Col>
-      </Form.Group>
-      {auth.user?.is_editor && (
-        <Form.Group as={Row} controlId="isEditor">
+    <>
+      {successMessage && (
+        <Alert
+          variant="success"
+          dismissible
+          onClose={() => {
+            setSuccessMessage(undefined);
+          }}
+        >
+          {successMessage}
+        </Alert>
+      )}
+      <Form id={props.formId} onSubmit={props.context.handleSubmit(submit)}>
+        <Form.Group as={Row} controlId="username">
+          <Form.Label column sm={2}>
+            Username
+          </Form.Label>
+          <Col sm={10}>
+            <Field
+              errors={props.context.errors}
+              type="text"
+              name="username"
+              disabled={props.user !== undefined}
+              validMessage={"This username is valid."}
+              isValid={
+                props.context.formState.dirtyFields.has("username") &&
+                !props.context.errors["username"]
+              }
+              ref={props.context.register({
+                maxLength: {
+                  value: 150,
+                  message: "USER.USERNAME.TOO_LONG",
+                },
+                required: true,
+                validate: async (username: string) => {
+                  if (props.user !== undefined) {
+                    // if we're editing a user the field is disabled anyway, don't validate
+                    return;
+                  }
+                  if (!(await validateUserNameDebounced(username))) {
+                    return Promise.resolve("USER.USERNAME.ALREADY_EXISTS");
+                  }
+                },
+              })}
+            />
+          </Col>
+        </Form.Group>
+        <Form.Group as={Row}>
+          <Form.Label column sm={2}>
+            Name
+          </Form.Label>
+          <Col sm={5}>
+            <Field
+              errors={props.context.errors}
+              type="text"
+              name="first_name"
+              placeholder="First Name"
+              ref={props.context.register({
+                maxLength: {
+                  value: 30,
+                  message: "USER.FIRST_NAME.TOO_LONG.30",
+                },
+                required: true,
+              })}
+            />
```

Comment:
On further testing, I'm able to just tab over this field without it triggering and I'm not sure why. I think it has something to do with validation being onChange now (ugh). Here's my stab at a possible solution:

- Change validation back to being onBlur
- For only the username and password fields, force validation onChange

This way all fields are validated even if they're just tabbed over, and we still keep the live validation for the two fields that need it most.

Reply:
This might just fix a bug I've been struggling with too, so I'll keep you posted.

Reply:
Well, it didn't fix my bug, but you can no longer tab past the first name field without being harassed about it.

https://github.com/UWmathNEWS/slugline-web/pull/13/commits/49fe676c2588c88a1f3379dbfa302c64c7ecc1a9